### PR TITLE
[PICA-29] transfer extrinsic for fnft + tests + benchmark

### PR DESCRIPTION
**Pull request by kkast**
https://app.clickup.com/t/20465559/PICA-29
1) Added extrinsic for transferring fnft
2) added becnhmarking
3) modified nonfungibles/transfer.rs tests to use Transfer trait transfer function
4) reused and modified Transfer trait tests for transfer extrinsic
5) hardcoded dali weights

### CODE REVIEW

File: code/parachain/frame/fnft/src/lib.rs [hunk 1]
```diff
@@ -199,6 +208,25 @@ pub mod pallet {
...
+		#[pallet::weight(T::WeightInfo::transfer())]
+		pub fn transfer(
+			origin: OriginFor<T>,
+			collection: FinancialNftCollectionIdOf<T>,
+			instance: FinancialNftInstanceIdOf<T>,
+			destination: AccountIdOf<T>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let fnfts = OwnerInstances::<T>::get(&who).ok_or(Error::<T>::MustBeOwner)?;
+			ensure!(fnfts.contains(&(collection, instance)), Error::<T>::MustBeOwner);
```
Comment (vimukthi-git) [inline code/parachain/frame/fnft/src/lib.rs:223]:
```suggestion
			ensure!(fnfts.contains(&(collection, instance)), Error::<T>::InstanceNotFound);
```

Comment (kkast) [reply code/parachain/frame/fnft/src/lib.rs:223]: should i change the error to InstanceNotFound to the line above as well then? There is MustBeOwner Error  defined in pallet but it is not used anywhere.

Comment (vimukthi-git) [reply code/parachain/frame/fnft/src/lib.rs:223]: `MustBeOwner` makes sense for the previous line no?

Comment (vimukthi-git) [reply code/parachain/frame/fnft/src/lib.rs:223]: Correct, I read it wrong.

**Review by vimukthi-git — APPROVED** [review]
LGTM, just minor testing related comments

File: code/parachain/frame/fnft/src/benchmarking.rs [hunk 2]
```diff
@@ -0,0 +1,30 @@
+use super::*;
+use crate::Pallet as Fnft;
+use codec::Encode;
+use frame_benchmarking::{account, benchmarks, impl_benchmark_test_suite};
+use frame_support::traits::tokens::nonfungibles::{Create, Mutate};
+use frame_system::RawOrigin;
+
+benchmarks! {
+  where_clause { where T::BlockNumber: From<u32>, T::FinancialNftCollectionId: From<u128>, T::FinancialNftInstanceId: From<u64> }
```
Comment (benluelo) [inline code/parachain/frame/fnft/src/benchmarking.rs:9]: please format this line (will have to be done manually unfortunately)

File: code/parachain/frame/fnft/src/benchmarking.rs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+		Fnft::<T>::mint_into(&collection_id, &created_nft_id, &user1)?;
+
+		Fnft::<T>::set_attribute(
+			&collection_id,
+			&created_nft_id,
+			&1_u32.encode(),
+			&1_u32.encode()
+		)?;
+	}: {
+		Fnft::<T>::transfer(
```
Comment (benluelo) [inline code/parachain/frame/fnft/src/benchmarking.rs:25]:
```suggestion
		_(
```

File: code/parachain/frame/fnft/src/benchmarking.rs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+			RawOrigin::Signed(user1).into(), collection_id, created_nft_id, user2
+		).expect("Transfer failed")
+	}
```
Comment (benluelo) [inline code/parachain/frame/fnft/src/benchmarking.rs:28]:
```suggestion
	}: _(RawOrigin::Signed(user1).into(), collection_id, created_nft_id, user2)
```

File: code/parachain/frame/fnft/src/test/nonfungibles/extrinsic.rs [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+		mock::{new_test_ext, Event, MockRuntime, Nft, Origin, Proxy},
+		prelude::{TEST_COLLECTION_ID, *},
+		ALICE, BOB, CHARLIE,
+	},
+	AccountIdOf, FinancialNftInstanceIdOf, Instance, OwnerInstances, Pallet,
+};
+
+/// Tests a simple transfer between 2 accounts, with only 1 total NFT existing.
+#[test]
+fn simple() {
```
Comment (benluelo) [inline code/parachain/frame/fnft/src/test/nonfungibles/extrinsic.rs:23]: either rename this to `transfer_simple` (or something similar) or move it into a new module `transfer`

File: code/parachain/frame/fnft/src/test/nonfungibles/extrinsic.rs [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+
+		process_and_progress_blocks::<Pallet<MockRuntime>, MockRuntime>(10);
+
+		assert_eq!(
+			Nft::owner(&TEST_COLLECTION_ID, &created_nft_id),
+			Some(ALICE),
+			"owner before transfer should be ALICE"
+		);
+
+		assert_ok!(Nft::transfer(Origin::signed(ALICE), TEST_COLLECTION_ID, created_nft_id, BOB));
```
Comment (benluelo) [inline code/parachain/frame/fnft/src/test/nonfungibles/extrinsic.rs:35]: use `RuntimeTrait::assert_extrinsic_event` to assert the event as well

File: code/parachain/frame/fnft/src/test/nonfungibles/extrinsic.rs [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+	});
+}
+
+/// Tests that an NFT that doesn't exist can't be transferred.
+#[test]
+fn instance_not_found() {
+	new_test_ext().execute_with(|| {
+		assert_noop!(
+			Nft::transfer(Origin::signed(ALICE), TEST_COLLECTION_ID, 1, BOB),
+			DispatchError::from(crate::Error::<MockRuntime>::MustBeOwner)
```
Comment (benluelo) [inline code/parachain/frame/fnft/src/test/nonfungibles/extrinsic.rs:255]: `DispatchError::from` is not required

**Review by vimukthi-git — APPROVED** [review]

**Review by benluelo — APPROVED** [review]

**Pull request merged**